library/psa_crypto.c

```diff
@@ -5735,97 +5735,33 @@ psa_status_t psa_key_derivation_input_key(
 /****************************************************************/
 /* Key agreement */
 /****************************************************************/
+#define PSA_KEY_AGREEMENT_MAX_SHARED_SECRET_SIZE MBEDTLS_ECP_MAX_BYTES
```

Comment:
This definition was in psa_crypto.c before my changes, the diff shows it because I just moved it while restructuring the code. It is used in psa_key_agreement_internal()

Reply:
But changing this goes with adding driver support. `MBEDTLS_ECP_MAX_BYTES` is only valid for ECDH (only implemented algorithm, so ok) for the curves defined by the software implementation (so not ok).

Which makes me realize that the definition of `PSA_VENDOR_ECC_MAX_CURVE_BITS` needs updating as well. And that's actually a bug already impacting ECDSA, so I'll make a small PR just for that.

Reply:
So to confirm, would the correct definition be 

```
#define PSA_KEY_AGREEMENT_MAX_SHARED_SECRET_SIZE PSA_RAW_KEY_AGREEMENT_OUTPUT_MAX_SIZE
```

Reply:
Or remove `PSA_KEY_AGREEMENT_MAX_SHARED_SECRET_SIZE` altogether.